base/client.jl

```diff
@@ -620,11 +620,11 @@ const main = MyApp.main
 """
 macro main(args...)
     if !isempty(args)
-        error("USAGE: `@main` is expected to be used as `(@main)` without macro arguments.")
+        error("`@main` is expected to be used as `(@main)` without macro arguments.")
```

Comment:
In the interest of taking advantage of semantic errors, can this `throw(ArgumentError(..))` instead?

Reply:
It's not really consistent across Base at the moment. When I reworked the error messages in Printf, I opted for `ArgumentError` because the error is about the arguments of a macro (see [here](https://github.com/JuliaLang/julia/blob/2a8bdd0812f621b38fd11411c99f357b4bc0b402/stdlib/Printf/src/Printf.jl#L1014)). I guess this could change in the future though if/when macros get the fancy error messages Claire talked about at JuliaCon. I'll keep this in mind until then though.